File: comuni/comuni-gui/src/shared/services/conversation/Answer.ts
import {Sentence} from './Sentence';
import {Question} from './Question';
export class Answer {

  sentences: any[];
  youCan: any[];

  addSentence(sentence: Sentence) {
    this.sentences.push(sentence);
  }

  getPossibleQuestions(): any[] {
    let questions = new Array<Question>();
    for (var question of this.youCan) {
      if (question.hasOwnProperty('ask')) {
        questions.push(new Question(question['ask']['question']));
      }
    }
    return questions;
  }

}

File: comuni/comuni-ws-lib/src/main/java/foundation/softwaredesign/comuni/lib/boundary/Answer.java
package foundation.softwaredesign.comuni.lib.boundary;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import javax.xml.bind.annotation.*;

/**
 * @param <T>
 * @author Lucas Reeh <lreeh@tugraz.at>
 */
@XmlRootElement(name = "answer")
@XmlAccessorType(XmlAccessType.FIELD)
public class Answer<T extends AbstractSentence> implements Serializable {

  private static final long serialVersionUID = 1L;

  public List<T> sentences = new ArrayList<>();

  @XmlElementWrapper(name = "youCanAsk")
  public List<Question> questions = new ArrayList<>();

  /**
   * zero argument constructor for jaxb
   */
  public Answer() {
    // zero argument constructor for jaxb
  }

  public void addPossibleQuestion(Question question) {
    this.questions.add(question);
  }

  public void addSentence(T sentence) {
    this.sentences.add(sentence);
  }

}
